pkg/openapi.go

```diff
@@ -1048,19 +1048,29 @@ func (ctx *resourceContext) propertyTypeSpec(parentName string, propSchema opena
 		}
 	}
 
+	// OpenAI shenanigans.
+	if val, has := propSchema.Value.Extensions["x-oaiTypeLabel"]; has && val == "string" {
+		return &pschema.TypeSpec{Type: "string"}, false, nil
+	}
```

Comment:
Can you please give me a couple of examples where `pulschema` failed to convert a property type? If it's something we can fix in `pulschema`, I would much prefer that as opposed to adding workarounds for specific extensions. Personally, I am hesitant to add this workaround here. Instead, I'd recommend handling this in the provider to fix up the OpenAPI spec before passing it to `pulschema`. For example, the Render provider required some fixes: https://github.com/cloudy-sky-software/pulumi-render/blob/praneetloke/updates/provider/pkg/gen/openapi_fixes.go#L37

Reply:
Sure, I understand your wish to keep the core clean of service-specific fixes. Will take it out.

The spec that made me put that in is the main OpenAI spec [here](https://raw.githubusercontent.com/openai/openai-openapi/master/openapi.yaml). Search for `x-oaiTypeLabel` and you'll find plenty of properties that have it instead of `type`.

Reply:
> The spec that made me put that in is the main OpenAI spec [here](https://raw.githubusercontent.com/openai/openai-openapi/master/openapi.yaml). Search for x-oaiTypeLabel and you'll find plenty of properties that have it instead of type.

Looking at a few of these, it seems that they use `x-oaiTypeLabel` with `anyOf` when a property has some pre-defined enum values but they also want to allow custom strings. I would expect `pulschema` to handle the `anyOf` definition instead of looking at the custom property anyway.

I have run into the situation with `anyOf` before with the [Tailscale Native Provider](https://github.com/cloudy-sky-software/pulumi-tailscale-native/blob/main/provider/cmd/pulumi-gen-tailscale-native/openapi.yml#L364-L368) too. The reason is that I don't think Pulumi's own schema has a way to represent an `anyOf` definition, unless I am wrong about that?

In any case, does `pulschema` not set the type for those properties because of that? Just wondering what the failure mode is today and if that can be improved with some sane defaults when handling an `anyOf` definition.